src/Serilog.Extensions.Hosting/Extensions/Hosting/DiagnosticContextCollector.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using Serilog.Events;
+
+namespace Serilog.Extensions.Hosting
+{
+    /// <summary>
+    /// A container that receives properties added to a diagnostic context.
+    /// </summary>
+    public sealed class DiagnosticContextCollector : IDisposable
+    {
+        readonly AmbientDiagnosticContextCollector _ambientCollector;
+        List<LogEventProperty> _properties = new List<LogEventProperty>();
+
+        internal DiagnosticContextCollector(AmbientDiagnosticContextCollector ambientCollector)
```

Comment:
Not ideal that this is `internal`, but it does save exposing some otherwise useless types.